### PR TITLE
[studio-ui] Path navigator update

### PR DESCRIPTION
**Pull request by aachavarria**
[studio-ui] Level descriptor as a child of an article doesn't list in the sidebar as expected

https://github.com/craftercms/craftercms/issues/4945

### CODE REVIEW

File: ui/app/src/state/epics/pathNavigator.ts [hunk 1]
```diff
@@ -57,6 +57,8 @@ export default [
           pathNavigatorFetchParentItems({
             id,
             path: storedState ? storedState.currentPath : payload.path,
+            ...(storedState?.offset && { offset: storedState.offset }),
```
Comment (rart) [inline ui/app/src/state/epics/pathNavigator.ts]: storedState.offset being 0 would result in a false which shouldn't

File: ui/app/src/components/PathNavigator/PathNavigator.tsx [hunk 2]
```diff
@@ -167,6 +168,12 @@ export default function PathNavigator(props: PathNavigatorProps) {
   const siteLocales = useSiteLocales();
   const hasActiveSession = useSelection((state) => state.auth.active);
 
+  useEffect(() => {
+    if (nnou(state?.keyword)) {
+      setKeyword(state.keyword);
```
Comment (rart) [inline ui/app/src/components/PathNavigator/PathNavigator.tsx:173]: Could we use state.keyword and not have a duplicate local state keyword var?

File: ui/app/src/state/reducers/pathNavigator.ts [hunk 3]
```diff
@@ -164,6 +170,10 @@ const reducer = createReducer<LookupTable<PathNavigatorStateProps>>(
           itemsInPath: children.map((item) => item.path),
           levelDescriptor: children.levelDescriptor?.path ?? null,
           breadcrumb: getIndividualPaths(withoutIndex(currentPath), withoutIndex(rootPath)),
+          leaves: [
+            ...state[id].leaves,
+            ...children.filter((item) => item.childrenCount === 0).map((item) => item.path)
```
Comment (rart) [inline ui/app/src/state/reducers/pathNavigator.ts]: Create a local util fn that receives children and returns an array of leaf paths and use it across all these instances

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/PathNavigator/PathNavigator.tsx (hunk 2 above)
Comment (aachavarria) [reply ui/app/src/components/PathNavigator/PathNavigator.tsx:173]: no at all, we need to keep one keyword value updated(the one the user should see in the pathNav) and another one debounced( the one is sent to the epics to fetch the data)

**Review by rart — APPROVED** [review]

**Pull request merged**